Tasks/NuGetCommand/nugetpublisher.ts

```diff
@@ -269,19 +270,20 @@ async function publishPackageVstsNuGetPushAsync(packageFile: string, options: IV
         vstsNuGetPushTool.arg(["-Verbosity", "Detailed"]);
     }
 
-    let exitCode: number = await vstsNuGetPushTool.exec();
-    if (exitCode === 0)
+    let execResult: IExecSyncResult = vstsNuGetPushTool.execSync();
+    if (execResult.code === 0)
     {
         return;
     }
 
     // ExitCode 2 means a push conflict occurred
-    if (exitCode === 2 && options.settings.continueOnConflict)
+    if (execResult.code === 2 && options.settings.continueOnConflict)
     {
         tl.debug(`A conflict ocurred with package ${packageFile}, ignoring it since "Allow duplicates" was selected.`)
         return;
     }
 
+    telemetry.logStderr(execResult.code, execResult.stderr);
     throw new Error(tl.loc("Error_UnexpectedErrorVstsNuGetPush"));
```

Comment:
We could throw stderr to the user here too.

Reply:
Fixed